### ISSUE TITLE
rfe: add method 'ReplaceDevice' to btrfs plugin

### ISSUE THREAD

**Issue by cmurf** [storaged-project/udisks#803]
http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Filesystem.BTRFS.html

Instead of AddDevice followed by RemoveDevice, it's preferred to use `btrfs replace`. Both `btrfs device add/remove` imply file system resize, and also a relatively expensive balance. Whereas replace is a variation on scrub, and is a sort of virtual and temporary raid1 between the device being replaced and the new device. Both devices are used for reads and writes, so there's better safety as well as performance.

There are two current caveats: the replacement block device must be equal to or larger than the device being replaced; and once replacement is complete, there is no automatic resize (grow) to take advantage of space for a larger device. 

It's actually more expedient to do a minimal resize (shrink) of a device being replaced, in order to make it possible to use `replace` instead of add followed by remove.
```
man btrfs replace
       start [-Bfr] <srcdev>|<devid> <targetdev> <path>
```

**Comment by tbzatek** [comment]
(marking as `help wanted` to encourage the community to come up with patches)

**Comment by vojtechtrefny** [comment]
Keep in mind that this needs to be added to [libblockdev](https://github.com/storaged-project/libblockdev) btrfs plugin first.

**Comment by cmurf** [comment]
Is it possible to reassign this to libblockdev? Or does it need a new issue opened?

**Comment by vojtechtrefny** [comment]
> Is it possible to reassign this to libblockdev? Or does it need a new issue opened?

GitHub really needs a "duplicate issue" button: https://github.com/storaged-project/libblockdev/issues/573
